app/app.go

```diff
@@ -434,32 +446,41 @@ func NewWasmApp(
 		app.bankKeeper,
 		scopedTransferKeeper,
 	)
-
 	transferModule := transfer.NewAppModule(app.transferKeeper)
 	transferIBCModule := transfer.NewIBCModule(app.transferKeeper)
 
-	// TODO: enable ICA something like this
-
-	// icaModule := ica.NewAppModule(&app.icaControllerKeeper, &app.icaHostKeeper)
-
-	// // NOTE: the IBC mock keeper and application module is used only for testing core IBC. Do
-	// // not replicate if you do not need to test core IBC or light clients.
-	// mockModule := ibcmock.NewAppModule(&app.ibcKeeper.PortKeeper)
-	// mockIBCModule := ibcmock.NewIBCModule(&mockModule, ibcmock.NewMockIBCApp(ibcmock.ModuleName, scopedIBCMockKeeper))
-
-	// app.icaControllerKeeper = icacontrollerkeeper.NewKeeper(
-	// 	appCodec, keys[icacontrollertypes.StoreKey], app.getSubspace(icacontrollertypes.SubModuleName),
-	// 	app.ibcKeeper.ChannelKeeper, // may be replaced with middleware such as ics29 fee
-	// 	app.ibcKeeper.ChannelKeeper, &app.ibcKeeper.PortKeeper,
-	// 	scopedicaControllerKeeper, app.MsgServiceRouter(),
-	// )
+	_ = app.getSubspace(icahosttypes.SubModuleName)
+	app.icaHostKeeper = icahostkeeper.NewKeeper(
+		appCodec,
+		keys[icahosttypes.StoreKey],
+		app.getSubspace(icahosttypes.SubModuleName),
+		app.ibcKeeper.ChannelKeeper,
+		&app.ibcKeeper.PortKeeper,
+		app.accountKeeper,
+		scopedICAHostKeeper,
+		app.MsgServiceRouter(),
+	)
+	app.icaControllerKeeper = icacontrollerkeeper.NewKeeper(
+		appCodec,
+		keys[icacontrollertypes.StoreKey],
+		app.getSubspace(icacontrollertypes.SubModuleName),
+		app.ibcKeeper.ChannelKeeper, // may be replaced with middleware such as ics29 fee
+		app.ibcKeeper.ChannelKeeper,
+		&app.ibcKeeper.PortKeeper,
+		scopedICAControllerKeeper,
+		app.MsgServiceRouter(),
+	)
+	icaModule := ica.NewAppModule(&app.icaControllerKeeper, &app.icaHostKeeper)
+	icaHostIBCModule := icahost.NewIBCModule(app.icaHostKeeper)
 
-	// // initialize ICA module with mock module as the authentication module on the controller side
+	// FIXME: this came from PR #793, but I don't like testing "mocks" in a real app
 	// icaAuthModule := ibcmock.NewIBCModule(&mockModule, ibcmock.NewMockIBCApp("", scopedICAMockKeeper))
-	// app.icaAuthModule = icaAuthModule
 
-	// icaControllerIBCModule := icacontroller.NewIBCModule(app.icaControllerKeeper, icaAuthModule)
-	// icaHostIBCModule := icahost.NewIBCModule(app.icaHostKeeper)
+	// TODO: clarify how to build icaAuthModule or remove icaController support?
```

Comment:
I am a bit lost here.
Can anyone show me an example of non-mock authorization for an icaController?

Otherwise, I would just make this a host... I want to integrate tried-and-true ibc code, not experiment here.

Reply:
Just host makes sense for me, in this PR